pkg/utils/download.go

```diff
@@ -48,33 +74,119 @@ func DownloadFromTarGzURL(URL string, destination string) error {
 	return err
 }
 
+func getURLToDownloadReleaseAsset(URL string, gitCredentials GitCredentials) (string, error) {
+	URLSlice := strings.Split(URL, "/")
+
+	domain := URLSlice[2]
+	client, err := getGitHubClient(domain, gitCredentials)
+	if err != nil {
+		return "", err
+	}
+
+	ctx := context.Background()
+
+	owner := URLSlice[3]
+	repo := URLSlice[4]
+	releases, gitHubResponse, err := client.Repositories.ListReleases(
+		ctx,
+		owner,
+		repo,
+		nil,
+	)
+	resp := gitHubResponse.Response
+	if resp.StatusCode != 200 {
+		return "", fmt.Errorf("[client.Repositories.ListReleases] GitHub responded with status code %d", resp.StatusCode)
+	}
+
+	release := URLSlice[7]
+	assetID, err := findAssetID(releases, release, URL)
+	if err != nil {
+		return "", err
+	}
+
+	_, redirectURL, err := client.Repositories.DownloadReleaseAsset(
+		ctx,
+		owner,
+		repo,
+		assetID,
+	)
+	if err != nil {
+		return "", fmt.Errorf("Repositories.DownloadReleaseAsset returned error: %v", err)
+	}
+	return redirectURL, nil
+}
+
+func findAssetID(releases []*github.RepositoryRelease, releaseName, URL string) (int64, error) {
+	for _, v := range releases {
+		if *v.TagName == releaseName {
+			for _, val := range v.Assets {
+				if *val.BrowserDownloadURL == URL {
+					return *val.ID, nil
+				}
+			}
+			return 0, fmt.Errorf("Cannot find matching assets for release %s", releaseName)
+		}
+	}
+	return 0, fmt.Errorf("Cannot find release %s", releaseName)
+}
+
 // DownloadFromRepoURL downloads a repo from a URL to a destination
-func DownloadFromRepoURL(repoURL string, destination string) error {
-	// expecting string in format 'https://github.com/<owner>/<repo>'
-	if strings.HasPrefix(repoURL, "https://") {
-		repoURL = strings.TrimPrefix(repoURL, "https://")
+func DownloadFromRepoURL(repoURL, destination string, gitCredentials GitCredentials) error {
+	// expecting string in format 'https://<github.com>/<owner>/<repo>'
+	URLSlice := strings.Split(repoURL, "/")
+
+	domain := URLSlice[2]
```

Comment:
We should probably check the length of URLSlice here too.

Reply:
Thanks, I'll add a test case in